src/Cli/Microsoft.DotNet.Cli.Utils/TypoCorrection.cs

```diff
@@ -12,35 +12,75 @@ namespace Microsoft.DotNet.Cli.Utils
 {
     public static class TypoCorrection
     {
+        private static readonly int s_minCurrentTokenLength = 3;
+        private static readonly int s_maxNumberOfSuggestions = 10;
+
         /// <summary>
-        /// Gets the list of tokens similar to <paramref name="currentToken"/>.
+        /// Gets the list of tokens similar to <paramref name="currentToken"/>
+        /// based on priority search:
+        /// 1. Starts with
+        /// 2. Contains - the call is restricted with <paramref name="currentToken"/> length check, minLength:3 and <param name="maxLevenshteinDistance">
+        /// 3. Levenshtein algorithm with <param name="maxLevenshteinDistance"> restriction
         /// </summary>
         /// <param name="possibleTokens">List of tokens to select from.</param>
         /// <param name="currentToken">The token that is being compared.</param>
         /// <param name="maxLevenshteinDistance">the difference between two strings, default: 3.</param>
         /// <returns>The enumerator to tokens similar to <paramref name="currentToken"/>.</returns>
         public static IEnumerable<string> GetSimilarTokens(IEnumerable<string> possibleTokens, string currentToken, int maxLevenshteinDistance = 3)
         {
-            int? bestDistance = null;
-            return possibleTokens
-                .Select(possibleMatch => (possibleMatch, distance: GetDistance(currentToken, possibleMatch)))
+            var possibleMatchesTuples = possibleTokens.Select(possibleMatch => (possibleMatch, distance: GetDistance(currentToken, possibleMatch)));
```

Comment:
performance wise i suggest to do it in this order:
- `StartsWith` matches
- if > 10, return first 10 ordered by distance
- add `Contains` matches (if >= 3 chars)
- if > 10, return first 10 ordered by rules
- do Levenshtein algorithm

Distance for `StartsWith` and `Contains` is: `candidate.Length - criteria.Length`.

Reply:
done
please take a look